src/librbd/io/ImageRequestWQ.cc

```diff
@@ -326,8 +348,12 @@ void ImageRequestWQ<I>::aio_write(AioCompletion *c, uint64_t off, uint64_t len,
 
   std::shared_lock owner_locker{m_image_ctx.owner_lock};
   if (m_image_ctx.non_blocking_aio || writes_blocked()) {
+    {
+      std::lock_guard locker{ordering_lock};
+      m_queued_or_blocked_io_tids.insert(tid);
```

Comment:
Nit: can this be moved to `queue`?

Reply:
Need those tids ordered in `unblock_flushes`